frontend/hooks/useRewardsHistory.ts

```diff
@@ -163,14 +168,67 @@ export const useRewardsHistory = () => {
       return parsedRewards.data;
     },
     refetchOnWindowFocus: false,
-    refetchInterval: ONE_DAY,
+    refetchInterval: ONE_DAY_IN_MS,
     enabled: !!serviceId,
   });
 
+  const latestRewardStreak = useMemo<number | null>(() => {
+    if (!data) return null;
+
+    // merge histories into single array
+    const allHistories = data.reduce(
+      (acc, { history }) => [...acc, ...history],
+      [] as EpochDetails[],
+    );
+
+    // remove all histories that are not earned
+    const earnedHistories = allHistories.filter((history) => history.earned);
+
+    // sort descending by epoch end time
+    const sorted = earnedHistories.sort(
+      (a, b) => b.epochEndTimeStamp - a.epochEndTimeStamp,
+    );
+
+    let streak = 0;
+    for (let i = 0; i < sorted.length; i++) {
+      const current = sorted[i];
+      // 1st iteration
+      if (i == 0) {
+        current.earned && streak++;
+        continue;
+      }
+
+      const previous = sorted[i - 1];
+
+      // 2nd iteration should consider that the
+      // first element may not have been earned yet
+      if (i == 1) {
+        if (!current.earned) break;
+        streak++;
+        continue;
+      }
+
+      // nth iterations should compare the time difference between epochs to detect streaks
+      if (!previous.earned) break;
+      if (!current.earned) break;
+
+      const epochGap = previous.epochEndTimeStamp - current.epochStartTimeStamp;
+
+      // if the gap between epochs is more than 1 day, break the loop
+      if (epochGap > ONE_DAY_IN_S) break;
```

Comment:
can we just not filter by earned and loop through all the elements, incrementing a counter until we reach an element marked as "not earned"? why do we need to check epochGap, if each epoch is ~a day?

Reply:
aaah see see